### PR TITLE
support new platform api, thermal and sfp part

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/psu.py [hunk 1]
```diff
@@ -26,30 +31,53 @@ def __init__(self, psu_index):
...
+    def _read_generic_file(self, filename, len):
+        """
+        Read a generic file, returns the contents of the file
+        """
+        result = 0
+        try:
+            with open(filename, 'r') as fileobj:
+                result = int(fileobj.read())
+        except:
+            logger.log_info("Fail to read file {}, maybe it doesn't exist".format(filename))
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/psu.py:69]: suggest rewording it, "file {} doesn't exist."

File: platform/mellanox/mlnx-platform-api/sonic_platform/psu.py [hunk 2]
```diff
@@ -60,15 +88,48 @@ def get_presence(self):
...
+
+        Returns:
+            A float number, the output voltage in volts, 
+            e.g. 12.1 
+        """
+        if self.psu_voltage is not None:
+            voltage = self._read_generic_file(os.path.join(self.psu_path, self.psu_voltage), 0)
+            return float(voltage) / 1000
+        else:
+            raise NotImplementedError
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/psu.py]: I am not sure whether we should raise "NotImplemenetedError" here, considering one case, if there is some bug with sysFS and the symbolic link was deleted, in this case maybe return a "none" would be better? just a thought.

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    },
+    #for 3700
+    {
+        THERMAL_DEV_CATEGORY_ASIC:(0,1),
+        THERMAL_DEV_CATEGORY_CPU_CORE:(0, 4),
+        THERMAL_DEV_CATEGORY_MODULE:(1, 32),
+        THERMAL_DEV_CATEGORY_PSU:(1, 2),
+        THERMAL_DEV_CATEGORY_CPU_PACK:(0,1)
+    },
+]
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py:236]: Add 3800?

File: platform/mellanox/mlnx-platform-api/sonic_platform/psu.py (hunk 2 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/psu.py]: it seems your suggestion is better. will fix it.
